### ISSUE TITLE
Question about pixel interval setting

### ISSUE THREAD

**Issue by kwea123** [JiayuYANG/CVP-MVSNet#2]
You claim that the depth search interval in the finer levels cannot be too narrow, otherwise the reprojected pixels will have similar features that make them difficult to distinguish. I agree with this statement, however a wider depth interval also makes the precision to decrease, doesn't it?

In my opinion, this problem comes down to **how large actually** is the depth interval. Let's take original MVSNet as example, it has 256 depth planes with an interval of 2mm. If your proposed method can narrow that down to, for example, 1mm in the finer levels, then it is indeed good, but if it only narrows to 1.9mm, it barely has no effect... why not just set a fixed value from the beginning? By the way, I also read a [paper](https://arxiv.org/pdf/1911.12012.pdf) that claims they can narrow the depth interval to 0.8mm with high precision.

So my questions are
1. How large actually is 0.5pix range in real mm?
2. If it's somewhere near the original 2mm, is it really effective/necessary to compute this? If it is <2mm, how small it is?

**Comment by MiaomiaoLiu** [comment]
Thank you for pointing us to this paper. 

Regarding your questions:
I would comment that the question should be more precise. 0.5 pix distance on 2D could mean a large interval on the visual ray if the centre point is far away from the camera or a very small range if the 3D point is close to the camera. It is also related to the camera parameters. I would answer your questions that it depends...

 It could be easily figured out based on some simple geometric derivation. We actually should put it online. Stay tuned.

**Comment by JiayuYANG** [comment]
Hi Kwea123,

Thank you for your interest in our project.

As mentioned by @MiaomiaoLiu, the depth interval depends on multiple things including image resolution, intrinsic & extrinsic parameters, pixel location on image, depth of the pixel, etc. 
Thus it really depends. 
You can refer to Figure 3 of the newest version of our paper on arXiv for more information.

However regarding your concern, I can provide you one example of the calculated interval.
For the first view of scan1 of the DTU dataset, during evaluation on 1600x1200 image using 5 levels, the calculated interval for 4 refinement levels of our model is:

| Level  |  Interval(mm) |
| ------------- | ------------- |
| 3 | 9.28 |
| 2 | 4.58 |
| 1 | 2.28 |
| 0 | 1.14 |

So the final depth interval is around 1.14mm for this view.
For other views it may vary based on above mentioned reasons.
Hope this address your concern.

Cheers,
Jiayu

**Comment by kwea123** [comment]
Based on my own experiments (not the implementation of your paper) I found that the performance first increases as the interval shrinks, then it starts to decrease when the interval becomes smaller than a threshold. I started from 2mm (uniform for all pixels) and gradually decrease it; the performance is the best at about 1.5mm then it starts to go worse.

I wanted to know if the optimal interval of yours aligns with mine, so I asked for the interval in mm. It turns out that we have roughly the same result: As the interval decreases we first get increasing performance, but there is a optimal value beyond which the performance will start to decrease (Table 4b). This threshold is about 0.5pix in your findings, which translates to about 1.~ mm.

Finally I'm still left with the question that how can [this paper](https://arxiv.org/abs/1911.12012) reach 0.8mm without performance drop? Will wait for them to opensource..
